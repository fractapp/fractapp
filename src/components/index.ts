export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
