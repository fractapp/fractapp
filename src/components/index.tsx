export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
