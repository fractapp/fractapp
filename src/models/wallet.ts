import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
