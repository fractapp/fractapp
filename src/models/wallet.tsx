import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
