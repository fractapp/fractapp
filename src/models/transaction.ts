import {Currency} from './wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export class Transaction {
  id: string;
  member: string;
  currency: Currency;
  value: number;
  txType: TxType;
  timestamp: number;
  fee: number;
  usdValue: number;
  usdFee: number;
  status: TxStatus;

  constructor(
    id: string,
    member: string,
    currency: Currency,
    txType: TxType,
    timestamp: number,
    value: number,
    usdValue: number,
    fee: number,
    usdFee: number,
    status: TxStatus,
  ) {
    this.id = id;
    this.member = member;
    this.currency = currency;
    this.value = value;
    this.txType = txType;
    this.timestamp = timestamp;
    this.fee = fee;
    this.usdValue = usdValue;
    this.usdFee = usdFee;
    this.status = status;
  }
}
