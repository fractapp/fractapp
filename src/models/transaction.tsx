import { ImageSourcePropType } from "react-native";
import { Currency } from "./wallet";

export enum TxType {
    None = 0,
    Sent,
    Received
}

export class Transaction {
    id: string;
    member: string;
    currency: Currency;
    value: number;
    txType: TxType;
    date: Date;

    usdValue: number;
    symbol: string;
    logo: ImageSourcePropType;

    constructor(id: string, member: string, currency: Currency, txType: TxType, date: Date, value: number) {
        this.id = id;
        this.member = member;
        this.currency = currency;
        this.value = value;
        this.txType = txType;
        this.date = date;
        //TODO get state
        let rate = 10
        this.usdValue = value * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
