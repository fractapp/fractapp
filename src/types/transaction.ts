import {Currency} from './/wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

/**
 * Transaction action type
 * @category Models
 */
export enum TxAction {
  Transfer,
  StakingReward,
  StakingCreateWithdrawalRequest,
  StakingWithdrawn,
  StakingOpenDeposit,
  StakingAddAmount,
  ConfirmWithdrawal,
  UpdateNomination
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export type Transaction = {
  id: string;
  hash: string;
  userId: string | null;

  address: string;
  currency: Currency;
  action: TxAction,
  txType: TxType;
  timestamp: number;

  value: number;
  planckValue: string;
  usdValue: number;
  fullValue: string;

  fee: number;
  planckFee: string;
  usdFee: number;

  status: TxStatus;
};


/**
 * @category Models
 */
export type ApiTransaction = {
  id: string;
  hash: string;
  currency: Currency;

  from: string;
  userFrom: string;

  action: TxAction,

  to: string;
  userTo: string;

  value: string;
  fee: string;
  price: number

  timestamp: number;
  status: TxStatus;
};

