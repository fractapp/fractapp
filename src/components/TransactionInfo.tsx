import React, { useState } from 'react';
import { StyleSheet, View, Image, Text, TouchableHighlight, Alert } from 'react-native';
import { Transaction, TxType } from '../models/transaction';
import { getSymbol, getLogo } from '../models/wallet'

/**
 * Component with transaction information 
 * @category Components
 */
export const TransactionInfo = ({ transaction, onPress }: { transaction: Transaction, onPress: () => void }) => {
    let prefix: string;
    let color: string;
    switch (transaction.txType) {
        case TxType.None:
            prefix = ""
            color = "#888888"
            break
        case TxType.Received:
            prefix = "+"
            color = "#84D371"
            break
        case TxType.Sent:
            prefix = "-"
            color = "#F45252"
            break
    }

    return (
        <TouchableHighlight
            onPress={onPress}
            key={transaction.id}
            underlayColor="#f8f9fb"
            style={styles.transaction}
        >
            <View style={{ width: "90%" }}>
                <View style={{ flex: 1, flexDirection: 'row' }}>
                    <Image
                        source={getLogo(transaction.currency)}
                        style={styles.logo}
                    />
                    <View style={{ flex: 1, flexDirection: 'column-reverse', marginLeft: 20, paddingTop: 5 }}>
                        <View style={{ flexDirection: 'row', paddingTop: 5 }} >
                            <View style={{ flex: 1 }}>
                                <Text style={[styles.balance, { textAlign: "left", color: color }]}>
                                    {prefix}{transaction.value} {getSymbol(transaction.currency)}
                                </Text>
                            </View>
                        </View>
                        <Text numberOfLines={1} style={styles.member}>
                            {transaction.member.length <= 10
                                ? `${transaction.member}`
                                : `${transaction.member.substring(0, 13)}...${transaction.member.substring(transaction.member.length - 13, transaction.member.length)}`}
                        </Text>
                    </View>
                </View>
            </View>
        </TouchableHighlight>
    );
}

const styles = StyleSheet.create({
    transaction: {
        paddingTop: 10,
        paddingBottom: 10,
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    member: {
        fontSize: 17,
        fontFamily: "Roboto-Regular",
        fontStyle: "normal",
        fontWeight: "normal",
        color: "black"
    },
    balance: {
        fontSize: 17,
        fontFamily: "Roboto-Regular",
        fontStyle: "normal",
    },
    logo: {
        width: 50,
        height: 50,
        borderRadius: 75,
        overflow: "hidden"
    }
});