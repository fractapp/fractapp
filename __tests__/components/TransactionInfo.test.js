import React from 'react';
import renderer from 'react-test-renderer';
import {TransactionInfo} from 'components/TransactionInfo';
import {Currency} from 'types/wallet';
import {TxStatus, TxType} from 'types/transaction';

jest.mock('utils/api', () => ({
  getImgUrl: jest.fn(),
}));

jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

it('Test one', () => {
  const tree = renderer
    .create(
      <TransactionInfo
        transaction={{
          id: '1',
          userId: 'userId',
          address: 'address#1',
          currency: Currency.DOT,
          txType: TxType.None,
          timestamp: new Date('12-12-2020').getTime(),
          value: 10,
          usdValue: 10,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Success,
        }}
        onPress={() => console.log('test')}
        user={null}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test two', () => {
  const tree = renderer
    .create(
      <TransactionInfo
        transaction={{
          id: '2',
          userId: 'userId',
          address: 'address#2',
          currency: Currency.KSM,
          txType: TxType.Sent,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 10,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Fail,
        }}
        onPress={() => console.log('test')}
        user={null}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test three', () => {
  const tree = renderer
    .create(
      <TransactionInfo
        transaction={{
          id: '3',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.KSM,
          txType: TxType.Received,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 10,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        onPress={() => console.log('test')}
        user={null}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test four', () => {
  const tree = renderer
    .create(
      <TransactionInfo
        transaction={{
          id: '4',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.KSM,
          txType: TxType.Received,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 10,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        onPress={() => console.log('test')}
        user={{
          id: 'id',
          name: 'name',
          username: 'username',
          avatarExt: 'png',
          lastUpdate: new Date('12-12-2020').getTime(),
          addresses: {
            0: 'addressOne',
            1: 'addressTwo',
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test five', () => {
  const tree = renderer
    .create(
      <TransactionInfo
        transaction={{
          id: '4',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.KSM,
          txType: TxType.Received,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 0,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        onPress={() => console.log('test')}
        user={{
          id: 'id',
          name: 'name',
          username: 'username',
          avatarExt: '',
          lastUpdate: new Date('12-12-2020').getTime(),
          addresses: {
            0: 'addressOne',
            1: 'addressTwo',
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test six', () => {
  const tree = renderer
    .create(
      <TransactionInfo
        transaction={{
          id: '4',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.KSM,
          txType: TxType.Received,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 0,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        onPress={() => console.log('test')}
        user={{
          id: 'id',
          name: '',
          username: 'username',
          avatarExt: '',
          lastUpdate: new Date('12-12-2020').getTime(),
          addresses: {
            0: 'addressOne',
            1: 'addressTwo',
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
