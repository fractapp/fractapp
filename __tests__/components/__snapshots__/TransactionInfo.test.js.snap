// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test five 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        height={50}
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          }
        }
        width={50}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 23,
            "bottom": 0,
            "height": 23,
            "justifyContent": "center",
            "left": 30,
            "position": "absolute",
            "width": 23,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F39B34",
                "fontSize": 23,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 10,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          name
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
              },
              Object {
                "color": "#67D44D",
                "textAlign": "left",
              },
            ]
          }
        >
          +
          10
           
          KSM
        </Text>
      </View>
      <View />
    </View>
  </View>
</View>
`;

exports[`Test four 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        height={50}
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          }
        }
        width={50}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 23,
            "bottom": 0,
            "height": 23,
            "justifyContent": "center",
            "left": 30,
            "position": "absolute",
            "width": 23,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F39B34",
                "fontSize": 23,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 10,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          name
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
              },
              Object {
                "color": "#67D44D",
                "textAlign": "left",
              },
            ]
          }
        >
          +
          10
           
          KSM
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
            },
            Object {
              "alignSelf": "center",
              "color": "#67D44D",
            },
          ]
        }
      >
        +
        $
        10
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test one 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/dot.png",
          }
        }
        style={
          Object {
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 0.8,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 10,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          address#1
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
              },
              Object {
                "color": "#888888",
                "textAlign": "left",
              },
            ]
          }
        >
          
          10
           
          DOT
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
            },
            Object {
              "alignSelf": "center",
              "color": "#888888",
            },
          ]
        }
      >
        
        $
        10
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test three 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/kusama.png",
          }
        }
        style={
          Object {
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 0.8,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 23,
            "bottom": 0,
            "height": 23,
            "justifyContent": "center",
            "left": 30,
            "position": "absolute",
            "width": 23,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F39B34",
                "fontSize": 23,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 10,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          1exaAg2VJRQ...o23R2T6EGdE
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
              },
              Object {
                "color": "#67D44D",
                "textAlign": "left",
              },
            ]
          }
        >
          +
          10
           
          KSM
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
            },
            Object {
              "alignSelf": "center",
              "color": "#67D44D",
            },
          ]
        }
      >
        +
        $
        10
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test two 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/kusama.png",
          }
        }
        style={
          Object {
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 0.8,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 23,
              "bottom": 0,
              "height": 23,
              "justifyContent": "center",
              "left": 30,
              "position": "absolute",
              "width": 23,
            },
            Object {
              "borderColor": "#EA4335",
              "borderWidth": 1,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#EA4335",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰅖
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "column",
            "marginLeft": 10,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "black",
              "fontFamily": "Roboto-Regular",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            }
          }
        >
          address#2
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "fontStyle": "normal",
              },
              Object {
                "color": "#EA4335",
                "textAlign": "left",
              },
            ]
          }
        >
          -
          10
           
          KSM
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 15,
              "fontStyle": "normal",
            },
            Object {
              "alignSelf": "center",
              "color": "#EA4335",
            },
          ]
        }
      >
        -
        $
        10
      </Text>
    </View>
  </View>
</View>
`;
